packages/transaction-controller/src/utils/simulation.ts

```diff
@@ -128,6 +136,11 @@ export async function getSimulationData(
       ],
       withCallTrace: true,
       withLogs: true,
+      ...(isReSimulatedDueToSecurity && {
+        blockOverrides: {
+          time: toHex(currentTimestampInSeconds + 60),
```

Comment:
As this is a client security concern and not pure simulation logic, could we decouple it from this util and pass in the `time` as an option instead so we calculate this within `TransactionController.ts` which already has the re-simulation logic?

Reply:
Done.